### DISCUSSION TITLE
OpenDTU-OnBattery & Victron VRM

### DISCUSSION THREAD

**Discussion by Jan-X** [hoylabs/OpenDTU-OnBattery#111] [Q&A]
Hallo,

Ich hab versucht, Euer Projekt an einen Victron EasySolar GX per VE.Direct anzuschließen, doch .... nix lief
(Es sollten so die Werte im VRM Portal zu sehen, und evtl auch den Hoymiles WR zu begrenzen (wenn der Akku voll ist.)

Zuerst dachte ich, es liegt an mir, doch als ich las dass Ihr nur RX Daten vom MPPT auswertet, doch dann las ich hier in der Q&A _"I use a serial connection (cable) to read the Ve.Direct protocol from my mppt battery charger"_  war  alles klar --> kann nicht funktionieren.
(Es wäre super, wenn Ihr das Projekt ein bisschen besser beschreiben könntet, hätte mir einiges an Zeit & Fehlersuche gespart)

**Frage:** 
ist es möglich, dass Ihr dieses Projekt so weit erweitert, dass

1) die OpenDTU  an ein Victron  Venus / GX die Daten per MQTT sendet (so dass man sie in deren VRM Portal (und lokal via Web) sehen kann)
Dazu müssten eigentlich  "nur" die entsprechenden OpenDTU Daten per MQTT richtig formatiert an rausgeschickt werden und sich die OpenDTU entsprechend dort an/ abmelden. 
Beschreibung siehe hier: [dbus-mqtt-devices](https://github.com/freakent/dbus-mqtt-devices)

2) Super wäre es, wenn sich die OpenDTU auch per MQTT mitbekommen würde, wann der Akku voll ist und dann entsprechend dies Solar-Leistung runterregelt.
 
**Weitere Beschreibungen/ Infos dazu hier:**
- [Generic driver for MQTT devices using dbus-mqtt ...](https://community.victronenergy.com/questions/121807/generic-driver-for-mqtt-devices-using-dbus-mqtt-db.html)
- [dbus-Mqtt](https://github.com/victronenergy/dbus-mqtt)

Danke

PS: [hier ](https://www.youtube.com/watch?v=UNuIOa72eP4)ist ein alternativer Weg beschrieben, doch der MQTT Weg ist sauberere, und er würde auch das Begrenzen der Wechselrichter Victron Venus / GX  per OpenDTU ermöglichen.

**Comment by helgeerbe** [comment]
Hallo @Jan-X ,
das mit der Dokumentation ist mir bewusst. Aber wir sind noch mitten in der Implementierung. Für dbus kannst du mal hier https://github.com/tbnobody/OpenDTU/discussions/349 und hier https://github.com/tbnobody/OpenDTU/discussions/217 schauen. Wenn es in der originalen OpenDTU verfügbar ist, werden wir es automatisch übernehmen.

Aktuell werten wir die die Batteriespannung, die am Inverter anliegt aus (oberer und unter Wert), bzw. vom MPPT das Load-Flag über VeDirekt. Ein weiteres interface kann ich mir vorstellen. Aber insgesamt muss erst mal alles laufen. 

Mitarbeit ist aber immer willkommen ;) Auch für die Dokumentation.

**Comment by Head** [comment]
Ich klemm mich hier mal mit rein, auch wenn ich kein Victron EasySolar GX habe, aber ein Venus OS auf dem Raspberry. Ich hätte jetzt den Pylo und den Laderegler mit dem Raspberry verkabelt und dann hätte das BMS ja mit dem Laderegler gesprochen. Den Hoymiles hätte ich dann irgendwie über NodeRed gescriptet.
Wenn ich dieses Projekt hier nehme liest der ESP von beiden mit und steuert den Hoymiles, aber nicht den Laderegler? Der Laderegler steuert dann rein über Spannung. Was hat das für Nachteile? Wäre doch im Grunde das gleiche wie wenn das Venus OS/Cerbo GX "abstürzen" würde.
Oder kann ich sogar ESP und Raspberry (sind ja Bus-Systeme?) gleichzeitig verkabeln?

ps. Ich kenn mich auch mit MQTT aus und kann programmieren, muss nur wissen wie und wo... Vielleicht kann ich mithelfen?

**Reply by helgeerbe** [reply]
Hallo @Head. Jetzt weiss ich nicht, ob ich dir ganz folgen kann.
* OpenDTU ist erstmal eine Bridge, die Daten vom Hoymiles lesen und Kommandos an den Hoymiles senden kann. 
* Über die Ve.Direct Schnittstelle lesen wir ebenfalls die Daten vom MPPT aus. 
* Über das CAN-Interface des Pylontech erhalten wir den Ladestand der Batterie (Alternativ und als Fallback, nehmen wir die Werte vom MPPT).
* Über den Shelly 3em erfahren wir, was gerade vom Netz im- bzw. exportiert wird.

Die OpenDTU kann erst mal alle Daten per MQTT bereitstellen.
Daneben verwenden wir die Daten um eine dynamisch Regelung des Hoymiles zu realisieren, um möglichst keinen Strom ins Netz zu exportieren.

Um Wandlungsverluste durch das Speichern in der Batterie zu verringern gibt es noch SolarPassThrough. Wir begrenzen dann den Hoymiles auf den Strom, der gerade vom MPPT geliefert werden kann. Nur wenn der MPPT mehr Strom liefert, als verbraucht wird, wird er in der Batterie gespeichert.

Warum sollten wir den Laderegler steuern? Am MPPT hängen die Panels. 

PS. Der Vorteil des ESPs ist, dass es einfach lokal (bis auf dem PowerMeter) läuft. Netzwerkausfälle stören den Regler erst mal nicht.

**Reply by Head** [reply]
Genau, soweit habe ich es verstanden und das klingt alles traumhaft.
Was ich nicht so ganz verstehe: Verkabel ich alles in der normalen Victron Welt miteinander, dann regelt doch das Pylontech BMS den MPPT? Oder der Multiplus, oder beide? Fehlt diese Kommunikation hier nicht?

**Reply by helgeerbe** [reply]
Der MPTT ist ein normales Ladegerät. Ich persönlich habe nur eine doofe Batterie. Der MPPT macht sie voll, wenn die Sonne scheint. Und wie jedes gute Ladegerät, schaltet er ab, sobald die Batterie voll ist. Bzw. geht in die Erhaltungsladung.

Der Hoymiles ist ein Verbraucher, der die Batterie leer saugt.  Der ESP regelt nun die Leistung, die der Hoymiles verbrauchen darf, bzw. schaltet ihn aus, wenn die Batterie leer ist.
Ich kann hier nur über die Batteriespannung abschätzen, wann die Batterie leer ist.

Ein BMS könnte ggf. genauere Ladeeinschätzungen geben.

Ve.direct von Victron wird nur dafür gebraucht um zu wissen, wie voll die Batterie ist. Und wie viel Leistung die Solarpannels gerade liefern.

**Reply by Head** [reply]
Zurück zu meiner Frage: Schließe ich das MPPT Ve.direct und den Pylontech CAN an ein Cerbo GX/Venus OS, dann regelt das BMS doch irgendwas im MPPT? Oder lese ich da irgend eine Magie aus dem Victron-Marketing raus, die gar nicht da ist?
Falls doch und ich nutze dieses Projekt, "fehlen" mir diese Ports, weil sie ja im ESP stecken und der MPPT muss "blind" steuern - wie bei deiner doofen Batterie. Deswegen: Kann ich nicht den CAN-bus an alle 3 anschließen? Pylontech, ESP und Cerbo. Das selbe mit dem MPPT. Oder, weil Ve.direct kein bus-system ist und nur 2 Geräte verbinden kann: Würde es dann nicht gut sein, wenn die openDTU diese Daten vom Victron MQTT bekommt? Wo ich (glaube ich) irgendwie beim Topic von Jan-X bin?

**Reply by Head** [reply]
z.B. die Antwort von Matthias hier: https://community.victronenergy.com/questions/55688/multiplus-ii-mit-pylontech-ohne-gx-moglich.html

> Ein GX Gerät (oder raspberry Pi mit Venus OS) ist nötig, da sich die Batterie ohne CAN Bus Kommunikation nach ein paar Minuten abschaltet.
> Zudem muss auch das Laden/Entladen der Batterie durch den Multi gestoppt werden, wenn es ein Problem mit einer Zelle gibt und das geht nur mit einer Datenverbindung zur Batterie.

Nach der Aussagen würde dieses Projekt hier ja gar nicht klappen.

Vielleicht weiß Berni2288 hier https://github.com/helgeerbe/OpenDTU-OnBattery/discussions/128 mehr dazu, er hat es ja scheinbar mit einer Pylontech am laufen und die scheint nicht abzuschalten.

**Reply by helgeerbe** [reply]
Schau mal hier

https://www.photovoltaikforum.com/thread/155456-pylontech-us2000c-und-victron-mppt-smartsolar-100-20-48v/

Aber ich denke @berni2288 kann dir hier von seiner Erfahrung berichten.

**Reply by berni2288** [reply]
Ich habe ein Pylontech US3000C und diese schaltet sich ohne CAN nicht ab. Vielleicht macht das der US2000 anders? Der US3000C jedenfalls nicht. Ich benutze die Pylontech ebenfalls als "dumme" Batterie und habe den monatelang ohne irgendwelche Kommunikation betrieben (also nur mit Kabel zum Wechselrichter).
Der einzige Grund warum ich die CAN-Schnittstelle implementiert hab ist um den SOC-Wert vom BMS auszulesen, weil der genauer und zuverlässiger ist als wenn man die Spannung für das Einschalten und Abschalten vom Wechselrichter nimmt.

Das BMS vom Pylontech hat bereits out of the box einige Sicherheitsfeatures eingebaut die meiner Meinung nach gut genug sind:

- Abschaltung wenn die Batterie leer ist (Falls aus irgendeinem Grund der Powerlimiter nicht mehr funktioniert)
- Verhinderung einer Ladung wenn die Batterie voll ist. Wobei mein Victron MPPT glaub ich schon bei 85% SOC aufhört zu laden, muss ich mir mal genauer ansehen).
- Soft-Start Funktion um die Kondensatoren vom Hoymiles am Anfang nicht zu überlasten.
- Temperaturüberwachung
- Zellenüberwachung

Beim Victron MPPT habe ich einfach das Profil für LifePO4 ausgewählt. Ob dieses optimal für die Pylontech ist weiß ich nicht, ich könnte mir vorstellen dass wenn man den Pylontech per CAN mit dem  Victron EasySolar GX verbindet, dass der Victron MPPT noch optimalere Laderegler-Werte für die Pylontech bekommt, z.B. dass er nicht bei 85% SOC aufhört zu laden.
Aber ich sehe die CAN-Schnittstelle jetzt eher als Komfort-Funktion, ich denke man kann die optimalen Laderegelereinstellungen auch genauso gut direkt am Victron MPPT einstellen.

Wenn etwas mit der Batterie nicht in Ordnung ist, dann sollte das Pylontech BMS eigentlich selbstständig notabschalten, ich sehe hier keinen Grund warum die Batterie dem Victron etwas mitteilen muss.

Allerdings betreibe ich selbst das ganze als Hobbyproject mit einem 800 Watt Hoymiles und 2 Solarpanels, bei einer 20 kWp Anlage würde ich ehrlich gesagt nicht dieses Projekt hier benutzen, da es bei so hohen Strömen und Spannungen ganz andere Sicherheitsanforderungen gibt, Wer eine professionelle Lösung will sollte sich woanders umschauen. Ich habe meine Batterie außerdem nicht  draußen stehen sondern im Warmen.

Einen Plan zum meinem Aufbau gibt es hier: https://github.com/helgeerbe/OpenDTU-OnBattery/discussions/128#discussion-4933876

**Reply by Head** [reply]
Das kann natürlich sein, dass der US2000 sich anders verhält. Der US3000C ist eh der beste im Preis pro kWh-Verhältnis :)
Ich sehe das auch als Hobby. Sobald es über den Microwechselrichter hinaus geht (Multiplus), läuft ja eh nichts mehr ohne eingetragenen Elektriker. Aber ich kenne auch mein Stromprofil (Grafana sei Dank) und mit 300W hätte ich vermutlich 80%, mit 400W 90% meiner kWh abgedeckt. Wozu also (im Sommer) 20kWp? (So viele Panels bekomme ich hier gar nicht unter.)

Die Sicherheitsfeatures vom Pylontech BMS sehe ich auch als genügend. Ich mach mir mehr Gedanken um die Akku-Lebenszeit, wenn irgendwie "falsch" geladen wird (wahrscheinlich unbegründet) und um die Effizienz vom System.

Ich habe mal meinen Verkäufer angerufen und der meinte, dass LiFePO4's nicht allein anhand der Volt den SoC anzeigen können. Die können 51V bei 80% oder bei 30% liefern. Stimmt so nicht 100%? Und er meinte, dass der Multiplus über den Cerbo dem MPPT mitteilt wie viel er zu liefern hat. Aha ok? Ich bin jetzt etwas verwirrt, bei einem Hobby möchte ich auch gerne etwas lernen, also dieses Video geschaut:
https://www.youtube.com/watch?v=svqEFpgjyDM
Ich nehme mit:
1. Sein Ladegerät macht beim Laden Pausen um die Ruhezellspannung zu messen. Ist das nicht schlecht, wenn gleichzeitig über den Hoymiles (wechselnde) Ströme gezogen werden, von denen der MPPT nix weiß?
2. Je schneller man lädt/entlädt, desto ungenauer ist die Berechnung des SoC über die Volt. Das ist schlecht, wenn der Victron MPPT ja über Volt laden muss?
3. Manche Laderegler und BMS zählen die Menge, die in und aus der Batterie fließt und ist der Spannungsbasierten Methode zu bevorzugen. Klingt so, als würde das Pylontech BMS so arbeiten. Aber dann weiß der Victron Laderegler ja gar nicht, wie der SoC aktuell ist.

Eins davon könnte deine 85% auf jeden Fall erklären.

Back to topic: **OpenDTU-OnBattery & Victron VRM**
Ich denke, es wäre in der Tat _besser_, wenn alles über einen Cerbo GX/Venus OS läuft und die OpenDTU als Wechselrichter integriert ist. Oder man alternativ rausfindet, wie und was Victron da noch so steuert und das hier dazu pflegt. Eine 20€ Alternative zum Cerbo GX wäre schon sexy ;)

Ich denke ich werde mal beides testen und alle Daten mitloggen. Also über Venus OS und über den ESP verkabeln. Vielleicht sehe ich dann schon einen Unterschied im Akku/SoC/Volt-Verhalten. Ich bin gespannt.
